code/game/gamemodes/objective.dm

```diff
@@ -955,6 +955,49 @@ GLOBAL_LIST_EMPTY(possible_items_special)
 				stolen_count++
 	return stolen_count >= 5
 
+/datum/objective/auto_complete
+	name="auto complete"
+	///This just stores the objective without the "<I>This objective auto-completes, so just have fun!</I>"
+	var/objective_text
+
+/datum/objective/auto_complete/New(text, antag_type = null)
+	make_random_objective(antag_type)
+	..()
+/datum/objective/auto_complete/proc/make_random_objective(antag_type = null, force_random_check = null)
+	var/objective
+	if((prob(50) || force_random_check) && antag_type)
+		switch(antag_type)
+			if("traitor") objective = pick(world.file2list("strings/objectives/antag/traitor.txt"))
+			if("wizard") objective = pick(world.file2list("strings/objectives/antag/wizard.txt"))
+			if("changeling") objective = pick(world.file2list("strings/objectives/antag/changeling.txt"))
+			if("BloodBrother") objective = pick(world.file2list("strings/objectives/antag/BloodBrother.txt"))
+	if(!objective) objective = pick(world.file2list("strings/objectives/antag/general.txt"))
+	explanation_text = "[objective] <I>This objective auto-completes, so just have fun!</I>"
+	objective_text = objective
+
+/datum/objective/auto_complete/update_explanation_text()
+	. = ..()
+	switch(objective_text)
```

Comment:
Objectives should probably be associative lists with a simple key so you dont have entire objective strings as switch statements.